tests/modules/test_text_processor.py

```diff
@@ -260,6 +274,139 @@ def test_subscribe_twice_hindi(self, texting_mock, logging_info_mock, logging_er
+        texting_mock.assert_called_with(message=first_response, phone_number="1-111-1114")
+        self.assertTrue(Contact.objects.filter(name="Rose", phone_number="1-111-1114",date_of_birth=datetime(2012, 10, 25, 0, 0), preg_update=False).exists())
+        self.assertTrue(t.get_contacts().exists())
+        t2 = TextProcessor(phone_number="1-111-1114")
+        second_response = t2.process("Born ROSE 25-11-2012")
+        self.assertTrue(t2.get_contacts().exists())
+        self.assertEqual(second_response, msg_subscribe("English").format(name="Rose"))
+        texting_mock.assert_called_with(message=second_response, phone_number="1-111-1114")
+        self.assertFalse(Contact.objects.filter(name="Rose", phone_number="1-111-1114",date_of_birth=datetime(2012, 10, 25, 0, 0), preg_update=False).exists())
+        self.assertFalse(Contact.objects.filter(name="Rose", phone_number="1-111-1114",date_of_birth=datetime(2012, 11, 25, 0, 0), preg_update=False).exists())
+        self.assertTrue(Contact.objects.filter(name="Rose", phone_number="1-111-1114", date_of_birth=datetime(2012, 11, 25, 0, 0), preg_update=True).exists())
+        self.assertTrue(t2.get_contacts().exists())
+        actual_groups = [str(g) for g in t2.get_contacts().first().group_set.all()]
+        expected_groups = ['Everyone - English', 'Text Sign Ups', 'Text Sign Ups - English']
+        self.assertEqual(actual_groups, expected_groups)
+
+    @patch("modules.text_processor.Texter.send")
+    def test_existing_contact_pregnancy_birthdate_update_english(self, texting_mock):
+        new_contact, _ = Contact.objects.update_or_create(name="Tina", phone_number="910003456789", language_preference="English", date_of_birth=datetime(2017, 7, 10, 0, 0), preg_signup=True)
+        t3 = TextProcessor(phone_number="910003456789")
+        self.assertTrue(t3.get_contacts().exists())
+        self.assertTrue(Contact.objects.filter(name="Tina", phone_number="910003456789", language_preference="English", date_of_birth=datetime(2017, 7, 10, 0, 0), preg_signup=True, preg_update=False).exists())
+        existing_contact_update = t3.process("Born Tina 25-07-2017")
+        self.assertEqual(existing_contact_update, msg_subscribe("English").format(name="Tina"))
+        texting_mock.assert_called_with(message=existing_contact_update, phone_number="910003456789")
+        self.assertFalse(Contact.objects.filter(name="Tina", phone_number="910003456789", language_preference="English", date_of_birth=datetime(2017, 7, 10, 0, 0), preg_signup=True, preg_update=False).exists())
+        self.assertFalse(Contact.objects.filter(name="Tina", phone_number="910003456789", language_preference="English", date_of_birth=datetime(2017, 7, 25, 0, 0), preg_signup=True, preg_update=False).exists())
+        self.assertTrue(Contact.objects.filter(name="Tina", phone_number="910003456789", language_preference="English", date_of_birth=datetime(2017, 7, 25, 0, 0), preg_signup=True, preg_update=True).exists())
```

Comment:
Can we just replace L347-349 with:

```
self.assertTrue(t3.get_contacts().first().preg_update)
self.assertTrue(t3.get_contacts().first().preg_signup)
```

Reply:
While those would pass, they wouldn't be as useful. The "BORN" texts update the contact's birthday, which this filtering checks on but that alternative wouldn't test. Also, if in the future we allow multiple phone numbers to a contact these tests would all need to be rewritten.